swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -293,6 +293,12 @@ final class IterableAPIInternal: NSObject, PushTrackerProtocol, AuthProvider {
                                  andFailureHandler: IterableAPIInternal.defaultOnFailure(identifier: "trackInAppClick"),
                                  forResult: apiClient.track(inAppClick: messageId, saveToInbox: saveToInbox, silentInbox: silentInbox, location: location, clickedUrl: clickedUrl, deviceMetadata: deviceMetadata))
     }
+    
+    func trackInAppDelivery(_ messageId: String, saveToInbox: Bool?, silentInbox: Bool?) {
```

Comment:
What do you think about moving the logic here? We'd just pass an `IterableInAppMessage` into this method and derive `saveToInbox` and `silentInbox` from it.

Reply:
Oooh, yes I like that abstraction.